session/handler.go

```diff
@@ -172,9 +177,18 @@ func (h *Handler) session(w http.ResponseWriter, r *http.Request, ps httprouter.
 		return
 	}
 
+	upgrade := r.URL.Query().Get("upgrade")
+	upgradeBool, err := strconv.ParseBool(upgrade)
+	if upgrade != "" && err != nil {
```

Comment:
Checking for empty string is unnecessary. `ParseBool()` will return ` strconv.ParseBool: parsing "": invalid syntax` in that case.
```suggestion
	if err != nil {
```

Reply:
no, this is expected to be empty if upgrade value is not set on the request so we need to handle this case separately